app/models/user_email.rb

```diff
@@ -17,6 +18,14 @@ class UserEmail < ActiveRecord::Base
 
   scope :secondary, -> { where(primary: false) }
 
+  def normalize_email
+    self.normalized_email = if self.email.present?
+      username, domain = self.email.split('@', 2)
+      username = username.gsub('.', '').gsub(/\+.*/, '')
+      "#{username}@#{domain}"
```

Comment:
It's not much use, but if we want to be pedant, we could even `downcase` the domain part of the email

```suggestion
      "#{username}@#{domain.downcase}"
```

Reply:
There is `before_validation :strip_downcase_email` which will do this before this method is called. What I mean is that self.email will be stripped and downcased before this method is called.

Reply:
Nice 👍